request_handler.py

```diff
@@ -0,0 +1,188 @@
+
+        :param asset_id: asset_id
+        :return: is_valid: Boolean True if valid, False if not.
+        """
+        if isinstance(asset_id, str) and asset_id.isalnum() and len(asset_id) <= 128:
+            return True
+        return False
+
+    def handle_request(self):
+        try:
+            GET = self._request.GET
+            asset_id = GET.get('state')
+            if not self.is_valid_asset_id(asset_id):
+                return self._return_error("Invalid asset id provided", 401)
+            self._rsh = RequestStateHandler(asset_id)
+            error = GET.get('error')
+            if error:
+                error_msg = GET.get('error_description')
+                return self._return_error(error_msg, 401)
+
+            code = GET.get('code')
+
+            ret_val, http_object = self._get_oauth_token(code)
+
+            if ret_val is False:
+                return http_object
+
+            return HttpResponse("You can now close this page", content_type="text/plain")
+        except Exception as e:
+            return self._return_error("Error handling request: {}".format(str(e)), 400)
+
+
+class RequestStateHandler:
+    def __init__(self, asset_id):
+        asset_id = str(asset_id)
+        if asset_id and asset_id.isalnum():
+            self._asset_id = asset_id
+        else:
+            raise AttributeError("RequestStateHandler got invalid asset_id")
+
+    def _encrypt_state(self, state):
+        try:
+            if 'oauth_token' in state:
+                token_list = ['access_token', 'id_token', 'refresh_token']
+                for token_name in token_list:
+                    if state['oauth_token'].get(token_name):
+                        state['oauth_token'][token_name] = encryption_helper.encrypt(  # pylint: disable=E1101
+                            state['oauth_token'][token_name],
+                            self._asset_id
+                        )
+        except Exception:
+            state.pop('oauth_token')
+            return state
+        return state
+
+    def _decrypt_state(self, state):
+        try:
+            if 'oauth_token' in state:
+                token_list = ['access_token', 'id_token', 'refresh_token']
+                for token_name in token_list:
+                    if state['oauth_token'].get(token_name):
+                        state['oauth_token'][token_name] = encryption_helper.decrypt(  # pylint: disable=E1101
+                            state['oauth_token'][token_name],
+                            self._asset_id
+                        )
+        except Exception:
+            message = "Error while decrypting"
+            state.pop('oauth_token', None)
+            return state, message
+        return state, ""
+
+    def _get_state_file(self):
+        dirpath = os.path.split(__file__)[0]
+        state_file = "{0}/{1}_state.json".format(dirpath, self._asset_id)
+        return state_file
+
+    def delete_state(self):
```

Comment:
Is there any point in returning True in this method if it always returns True? Same for 'save_state'

Reply:
These methods are exclusively employed for temporarily creating and storing code for OAuth operations. Therefore, we consistently return true here.

Reply:
Do we check/use this return value anywhere in the connector?

Reply:
I still believe that we can remove the return statement in this method if it isn't used at all, but I will leave the final decision up to you.